File: Spade.Database/Entities/GuildSettings.cs
using Canducci.MongoDB.Repository.MongoAttribute;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Spade.Database.Entities
{
	public interface IGuildSettings
	{
		ObjectId Id { get; init; }

		string GuildId { get; init; }
		string Prefix { get; init; }
	}

	[MongoCollectionName("GuildSettings")]
	public record GuildSettings : IGuildSettings
	{
		[BsonId]
		public ObjectId Id { get; init; }

		[BsonElement("guild_id")]
		[BsonRequired]
		public string GuildId { get; init; }

		[BsonElement("prefix")]
		[BsonRequired]
		public string Prefix { get; init; }
	}
}

File: internal/database/guild_settings.go
package database

import "encoding/json"

type GuildSettings struct {
	GuildID        string   `json:"guild_id"`
	EnabledModules []string `json:"enabled_modules"`
}

func (g *GuildSettings) String() string {
	b, _ := json.Marshal(g)
	return string(b)
}
